File: frontend/src/types/note.ts
export interface Note {
  id: string;
  user_id: string;
  workspace_id?: string;
  document_id?: string;
  title: string;
  content: string;
  created_at: string;
  updated_at: string;
}

export interface NoteCreate {
  title: string;
  content: string;
  workspace_id?: string;
  document_id?: string;
}

export interface NoteUpdate {
  title?: string;
  content?: string;
  workspace_id?: string;
  document_id?: string;
}

export interface GenerateNotesRequest {
  document_ids?: string[];
  workspace_id?: string;
}

File: backend/app/models/note.py
"""Note model."""

from datetime import datetime
import uuid

from sqlalchemy import Column, DateTime, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from app.db.base import Base


class Note(Base):
    """Manual note model."""

    __tablename__ = "notes"

    id = Column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, index=True
    )
    user_id = Column(
        UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )
    workspace_id = Column(
        UUID(as_uuid=True), ForeignKey("workspaces.id", ondelete="CASCADE"), nullable=True, index=True
    )
    document_id = Column(
        UUID(as_uuid=True), ForeignKey("documents.id", ondelete="CASCADE"), nullable=True, index=True
    )
    title = Column(String, nullable=False)
    content = Column(Text, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(
        DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False
    )

    user = relationship("User")
    workspace = relationship("Workspace")
    document = relationship("Document")
